demo/index.html

```diff
@@ -100,7 +107,16 @@
                       value: false
                     },
                     selected: String,
-                    steps: Array
+                    steps: Array,
+                    i18n: {
+                      type: Object,
+                      value() {
+                        return {
+                          'OPTIONAL': 'Optional',
+                          'step-3-key': 'Preview'
```

Comment:
No, you have to be extremely careful using shared keys.
i18n should be handled at a component level. By including all possible keys at an app level you are including a bunch of extra data that may not be needed.